src/pages/VenuePage/TemplateWrapper.tsx

```diff
@@ -143,16 +145,25 @@ const TemplateWrapper: React.FC<TemplateWrapperProps> = ({ venue }) => {
       template = <div>Unknown Template: ${(venue as AnyVenue).template}</div>;
   }
 
-  // @debt remove backButton from Navbar
+  const parentVenueId = venue?.parentId ?? "";
+
+  const backToParentVenue = useCallback(() => {
+    enterVenue(parentVenueId, { customOpenRelativeUrl: history.push });
+  }, [history.push, parentVenueId]);
+
   return (
     <RelatedVenuesProvider venueId={venue.id}>
       <ReactionsProvider venueId={venue.id}>
-        <WithNavigationBar
-          fullscreen={fullscreen}
-          hasBackButton={hasBackButton}
-        >
+        <WithNavigationBar fullscreen={fullscreen}>
           <AnnouncementMessage message={venue.bannerMessage} />
           {template}
+          {!TemplatesWithoutBackButton.includes(venue.template) &&
+            parentVenue && (
+              <BackButton
+                onClick={backToParentVenue}
+                title={`Back ${parentVenue ? ` to ${parentVenue.name}` : ""}`}
```

Comment:
This will end up with a double space in it currently.

If there is no `parentVenue`, will there be anything to link back to anyway? You're already conditionally rendering this whole block on the presence of `parentVenue`. So can probably just get rid of this ternary entirely.

Reply:
After extracting the condition above the JSX: https://github.com/sparkletown/sparkle/pull/1492#discussion_r647907979 This is crying that `parentVenue` might be `undefined`.

This works:
```tsx
{!TemplatesWithoutBackButton.includes(venue.template) && parentVenue && (
  <BackButton
     onClick={backToParentVenue}
     title={`Back to ${parentVenue.name}`}
   />
)}
```
This doesn't:
```tsx
const showBackButton = !TemplatesWithoutBackButton.includes(venue.template) && parentVenue
{showBackButton && (
  <BackButton
     onClick={backToParentVenue}
     title={`Back to ${parentVenue.name}`}
   />
)}
```

I can use optional `parentVenue?.name`. But this doesn't change the fact that typescript doesn't 'read' this correctly because I'm already checking if `parentVenue` exists. Any idea how to fix this?